src/libasr/codegen/asr_to_c.cpp

```diff
@@ -238,16 +244,26 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
                 if( is_array ) {
                     bool is_fixed_size = true;
                     dims = convert_dims_c(t->n_dims, t->m_dims, v_m_type, is_fixed_size, true);
-                    std::string encoded_type_name = "i" + std::to_string(t->m_kind * 8);
-                    bool is_struct_type_member = ASR::is_a<ASR::StructType_t>(
-                         *ASR::down_cast<ASR::symbol_t>(v.m_parent_symtab->asr_owner));
-                    generate_array_decl(sub, std::string(v.m_name), type_name, dims,
-                                        encoded_type_name, t->m_dims, t->n_dims,
-                                        use_ref, dummy,
-                                        (v.m_intent != ASRUtils::intent_in &&
-                                        v.m_intent != ASRUtils::intent_inout &&
-                                        v.m_intent != ASRUtils::intent_out &&
-                                        !is_struct_type_member) || force_declare, is_fixed_size);
+                    if( t->m_kind == 1 && v.m_abi == ASR::abiType::BindC && is_fixed_size ) {
+                        if( !force_declare ) {
+                            force_declare_name = std::string(v.m_name);
+                        }
+                        sub = "char " + force_declare_name + dims;
```

Comment:
Why not using `int8_t` ? It seems that's cleaner and consistent with i16, i32, i64, etc.

Reply:
I think in the issue it was mentioned to create `char buffer[64]` for `buffer: i8[64]`.